### ISSUE TITLE
clang-cl out of memory error

### ISSUE THREAD

**Issue by codethinki** [llvm/llvm-project#76625]
if you create an array in a function body with a very large size (idk string::npos size) it will crash, saying that it encountered an out of memory error and i should report it. 
msvc says "total size of array must not exceed 0x7fffffff bytes"

```cpp
constexpr uint32_t MIN_WORDS = 1;
constexpr uint32_t MAX_WORDS = 3;
vector<string> solve(const string_view letter_boxed_sides) {
     constexpr uint32_t DICTIONARIES = MIN_WORDS - MAX_WORDS;
     array<vector<string>, DICTIONARIES> dictionaries{};
     array<array<uint32_t, 27>, DICTIONARIES> dictionaryEntries{};
     ...
};
```

**Comment by zufuliu** [comment]
It no longer crashes (see issue #45080), https://godbolt.org/z/6vvqKKsE6

```
<source>:10:16: warning: stack frame size (566935682888) exceeds limit (4294967295) in 'solve[abi:cxx11](std::basic_string_view<char, std::char_traits<char>>)' [-Wframe-larger-than]
   10 | vector<string> solve(const string_view letter_boxed_sides) {
      |  
```

**Comment by codethinki** [comment]
hm maybe vs is outdated idk imma check later
